src/libponyc/codegen/codegen.c

```diff
@@ -512,7 +512,9 @@ void codegen_shutdown(pass_opt_t* opt)
 
 bool codegen(ast_t* program, pass_opt_t* opt)
 {
-  printf("Generating\n");
+  if(opt->verbosity >= VERBOSITY_DEFAULT) {
```

Comment:
Since these lines always compare `>=` and always use `printf`, to me, it seems like it would be nicer if this were a macro.  For example, perhaps something like:
```c
#define LOG(opt, level, args) if(opt->verbosity >= level) { printf args ; }

// ...

LOG(opt, VERBOSITY_DEFAULT, ("Generating\n"));

// ...

LOG(c->opt, VERBOSITY_DEFAULT, ("Linking %s\n", file_exe));
```

Although maybe @sylvanc will hit me over the head for such a suggestion, I don't know :stuck_out_tongue:.